core/src/main/java/org/apache/calcite/rex/RexUtil.java

```diff
@@ -2799,6 +2800,175 @@ private RexNode or(Iterable<? extends RexNode> nodes) {
     }
   }
 
+  /**
+   * Helper class that expands predicates from disjunctions (splited by K).
+   *
+   * @param <K> The dimension used to split predicates in disjunctions. If you want to expand
+   *           predicates that can be pushed down to a single table from the disjunction, K is
+   *           {@link RelTableRef}; if you want to expand predicates that can be pushed down to
+   *           Join inputs from the disjunction, K is string whose value is 'left' or 'right'.
+   */
+  public abstract static class ExpandDisjunctionHelper<K> {
+    private final RelBuilder relBuilder;
+
+    private final int maxNodeCount;
+
+    // Used to record the number of redundant expressions expanded.
+    private int currentCount;
+
+    public ExpandDisjunctionHelper(RelBuilder relBuilder, int maxNodeCount) {
+      this.relBuilder = relBuilder;
+      this.maxNodeCount = maxNodeCount;
+    }
+
+    public Map<K, RexNode> expand(RexNode condition) {
+      try {
+        this.currentCount = 0;
+        return expandDeep(condition);
+      } catch (OverLimitException e) {
+        return new HashMap<>();
+      }
+    }
+
+    /**
+     * Expand predicates recursively that can be pushed down to a single K.
+     *
+     * @param condition   Predicate to be expanded
```

Comment:
can you add a java doc for K?
I know it's present above, but it's worth repeating.